marketplace/plugins/supabase/lib/index.ts

```diff
@@ -0,0 +1,207 @@
+import { QueryError, QueryResult, QueryService, ConnectionTestResult } from '@tooljet-marketplace/common';
+import { createClient } from '@supabase/supabase-js';
+import { SourceOptions, QueryOptions, Column, Filter, Sort } from './types';
+
+export default class Supabase implements QueryService {
+  async run(sourceOptions: SourceOptions, queryOptions: QueryOptions, dataSourceId: string): Promise<QueryResult> {
+    const supabaseClient = await this.getConnection(sourceOptions);
+    const operation = queryOptions.operation;
+    let result = {};
+    let error: any = false;
+    try {
+      if (!operation) throw new Error('Select one operation');
+      const { get_table_name, create_table_name, update_table_name, delete_table_name, count_table_name } =
+        queryOptions;
+      const tableNameValues = {
+        get_rows: get_table_name,
+        create_row: create_table_name,
+        update_row: update_table_name,
+        delete_row: delete_table_name,
+        count_rows: count_table_name,
+      };
+      if (!tableNameValues[operation]) throw new Error('Table name is required');
+      let res: any;
+      switch (operation) {
+        case 'get_rows':
+          res = await this.getRows(queryOptions, supabaseClient);
+          error = res.error;
+          result = res.data;
+          break;
+        case 'create_row':
+          res = await this.createRows(queryOptions, supabaseClient);
+          error = res.error;
+          result = { created: true };
+          break;
+        case 'update_row':
+          res = await this.updateRows(queryOptions, supabaseClient);
+          error = res.error;
+          result = res.data;
+          break;
+        case 'delete_row':
+          res = await this.deleteRows(queryOptions, supabaseClient);
+          error = res.error;
+          result = { deleted: true };
+          break;
+        case 'count_rows':
+          res = await this.countRows(queryOptions, supabaseClient);
+          error = res.error;
+          result = res.data;
+          if (Array.isArray(result)) {
+            result = result[0];
+          }
+          break;
+        default:
+          break;
+      }
+    } catch (err) {
+      throw new QueryError('Query could not be completed', err.message, {});
+    }
+
+    if (error) {
+      throw new QueryError('Query could not be completed', error, {});
+    }
+
+    return {
+      status: 'ok',
+      data: result,
+    };
+  }
+
+  async getRows(queryOptions: QueryOptions, supabaseClient: any) {
```

Comment:
Also let's get rid of `any` usages, supabase has typescript classes defined. We can use them here.

Point to note here being, 
- let typescript infer class wherever possible
- if we have to declare types lean away from using `any`
- if you have no types available try to use `unknown` instead
- `any` should be used as the last resort

Let typescript make our dev lives easier and sane when type inferences are available :grinning:

Reply:
@akshaysasidrn This is done. Updated Types. I use `any` for third party packages (had some bad experiences in past with some packages 🙂). Let me know if any change required.